sympy/integrals/tests/test_integrals.py

```diff
@@ -309,7 +309,7 @@ def test_issue_3679():
 
 def test_issue_3686():  # remove this when fresnel itegrals are implemented
     from sympy import expand_func, fresnels
-    assert expand_func(integrate(sin(x**2), x)) == \
+    assert simplify(expand_func(integrate(sin(x**2), x))) == \
         sqrt(2)*sqrt(pi)*fresnels(sqrt(2)*x/sqrt(pi))/2
```

Comment:
Again simplify should not be used.

In these cases I think it's better to change the expected form of the output in the tests so that PR reviewers can see what has changed.

Reply:
I am not comfortable with the simplify either. It is due to the hyperexpand function getting a different result from a matrix multiplication which permutes what it was expecting. Still trying to find a workaround.

Reply:
Changed `hyperexpand()` matrix multiplication usage in 3 places, it was expecting `_eval_matrix_mul()` to always return the form `c11*b11 + c12*b21 + ...` but now it optimizes. Had three options, change `_eval_matrix_mul()` to not optimize on 1 row * 1 col, have `hyperexpand()` explicitly call `_eval_matrix_mul(expand=False, simplify=False)` or just remove the matrix multiplication and reduce in the code since the matrices are always 1row*1col (I checked them all and the very structure of the code implies this as it expects a single scalar answer), it is also faster and cleaner.

This returns clean answers like before and let me remove the simplify checks in the tests.

Reply:
Good work!